### PR TITLE
feat: Integrate folder picker on iPad - WPB-11895

### PR DESCRIPTION
**Pull request by samwyndham**
### Issue

This PR integrates the folder picker feature on **iPad ONLY** _hopefully_ meeting the acceptance requirements outline in https://wearezeta.atlassian.net/browse/WPB-11754.

https://github.com/user-attachments/assets/c580ecd9-19c9-4de2-9f0c-1d83f69f1675

### Testing

**Note**: Currently there is no way to add/remove folders on `iOS`. Instead use the web interface to add/remove folders as needed.

Please test on **iPad**.

#### Folders use case

0. Ensures your account has some folders in it.
1. Rotate to portrait
2. Tap the side bar button if the sidebar isn't already showing
3. Tap `Folders` from the sidebar.
4. Verify a popover view opens with a list of folders. No folder is selected.
5. Tap a folder.
6. Verify the popover dismisses and the selected folders contents are displayed.
7. Tap `Folders` from the sidebar.
8. Verify a popover view opens with a list of folders. The previously selected folder is still selected.
9. Tap `All` from the sidebar.
10. Verify all conversations are displayed.
11. Tap `Folders` from the sidebar.
12. Verify a popover view opens with a list of folders. No folder is selected.
13. Rotate to landscape and repeat steps 1 - 12

#### No folders use case

0. Ensure your account has no folders.
2. Tap the side bar button if the sidebar isn't already showing
3. Tap `Folders` from the sidebar.
4. Verify a popover view opens showing an empty state message.
5. Tap `How to add a conversation to a folder`.
6. App navigates to https://support.wire.com/hc/en-us/articles/360002855817-Add-a-conversation-to-a-custom-folder in Safari. 
7. Return to the app.
8. Close the popover

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [x] Make sure you use the API for UI elements that support large fonts.
- [x] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [x] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: WireUI/Sources/WireMainNavigationUI/Protocols/Sidebar/MainSidebarMenuItemRepresentable.swift [hunk 1]
```diff
@@ -18,19 +18,8 @@
```
Comment (samwyndham) [inline WireUI/Sources/WireMainNavigationUI/Protocols/Sidebar/MainSidebarMenuItemRepresentable.swift:18]: These changes are no longer necessary as my implementation has changed but this deleted code is not used. @caldrian Should I revert the change in this file.

Comment (caldrian) [reply WireUI/Sources/WireMainNavigationUI/Protocols/Sidebar/MainSidebarMenuItemRepresentable.swift:18]: Deleting it is fine.

File: WireUI/Sources/WireSidebarUI/Views/SidebarViewControllerDelegate.swift [hunk 2]
```diff
@@ -16,9 +16,12 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
+import Foundation
+
 @MainActor
 public protocol SidebarViewControllerDelegate: AnyObject {
     func sidebarViewControllerDidSelectAccountImage(_ viewController: SidebarViewController)
+    func sidebarViewController(_ viewController: SidebarViewController, didTapFoldersAt rect: CGRect)
```
Comment (caldrian) [inline WireUI/Sources/WireSidebarUI/Views/SidebarViewControllerDelegate.swift]: `didtapFoldersAt` rect sounds really fragile to me.
Is there no other way to know which folder was selected?

Comment (samwyndham) [reply WireUI/Sources/WireSidebarUI/Views/SidebarViewControllerDelegate.swift]: @caldrian Maybe this is badly named. This is not about identifying which folder was tapped. It is the frame of the `Folders` button in the sidebar.

File: WireUI/Sources/WireSidebarUI/Views/SidebarMenuItemView.swift [hunk 3]
```diff
@@ -87,6 +90,11 @@ struct SidebarMenuItemView<TitleView: View>: View {
             .cornerRadius(backgroundCornerRadius)
         }
         .dynamicTypeSize(...DynamicTypeSize.accessibility1)
+        .onGeometryChange(for: CGRect.self) { proxy in
```
Comment (caldrian) [inline WireUI/Sources/WireSidebarUI/Views/SidebarMenuItemView.swift]: I don't know what exactly happens here, but could it be that this causes performance overhead when scrolling?
I don't think every menu time view should keep track of its frame on the screen.
Maybe we can brainstorm for a bitter idea?

**Review by caldrian — APPROVED** [review]

File: WireUI/Sources/WireSidebarUI/Views/SidebarMenuItemView.swift (hunk 3 above)
Comment (samwyndham) [reply WireUI/Sources/WireSidebarUI/Views/SidebarMenuItemView.swift]: Updated so that we only track the frame for the `Folders`  menu item. I know it is not ideal but don't have a better solution right now.

File: WireUI/Sources/WireSidebarUI/Views/SidebarViewControllerDelegate.swift (hunk 2 above)
Comment (samwyndham) [reply WireUI/Sources/WireSidebarUI/Views/SidebarViewControllerDelegate.swift]: I'll rename this to make it more explicit so that we know this corresponds to a menu item, not a folder.

**Review by agisilaos — APPROVED** [review]
